### PR TITLE
feat: NPC driven tutorial quests

### PR DESCRIPTION
**Pull request by KnightNiwrem**
This pull request introduces a robust system for conditional dialogue choices and quest integration within NPC conversations. The main improvements include supporting dialogue choices that depend on game state (such as quest progress or skill levels), handling quest actions directly from dialogue, and ensuring tutorial quests are only started via NPCs. Comprehensive tests and data updates ensure correctness and a smooth player experience.

**Dialogue System Enhancements**

* Added support for conditional dialogue choices, allowing choices to appear or be selectable only if certain game state conditions (like quest state or skill level) are met. Introduced the `checkCondition` function and updated the dialogue engine to use it. [[1]](diffhunk://#diff-c4e46904276cf715e624cb7ebd957116e6b1e3c1018fa18ef36d1073e470a269R41-R98) [[2]](diffhunk://#diff-c4e46904276cf715e624cb7ebd957116e6b1e3c1018fa18ef36d1073e470a269L174-R240) [[3]](diffhunk://#diff-c4e46904276cf715e624cb7ebd957116e6b1e3c1018fa18ef36d1073e470a269R286-R301) [[4]](diffhunk://#diff-12f59ceb7110735449b2c1ff896dedc406dc4f4832d95740300ecdab6a29b238L61-R103) [[5]](diffhunk://#diff-12f59ceb7110735449b2c1ff896dedc406dc4f4832d95740300ecdab6a29b238L137-R184) [[6]](diffhunk://#diff-e53dc0a9824d043473e6cac8037454535f8329eabd633f2c2050cd17538a866cL33-R43)
* Dialogue choices can now trigger quest actions (start or complete quests) directly from the conversation UI, updating the game state accordingly. [[1]](diffhunk://#diff-12f59ceb7110735449b2c1ff896dedc406dc4f4832d95740300ecdab6a29b238L61-R103) [[2]](diffhunk://#diff-12f59ceb7110735449b2c1ff896dedc406dc4f4832d95740300ecdab6a29b238L137-R184) [[3]](diffhunk://#diff-e53dc0a9824d043473e6cac8037454535f8329eabd633f2c2050cd17538a866cR130-R250)

**Quest System Adjustments**

* Tutorial quests are now hidden from the general available quest list and can only be started via NPC dialogue, preventing confusion and guiding players through the intended flow. [[1]](diffhunk://#diff-6ca5003a11b26c78b815026e2c6b476efa9f92ff18809fcdd3c1fc56486055e4R118-R121) [[2]](diffhunk://#diff-b097dcb822033d093e324a83782c0fddfc8e532c11a357d555ccf4b287f00d51R84-R115)
* Updated the Oak NPC's dialogue tree to use the new conditional/actionable choices for starting and completing tutorial quests, with precise conditions for when each option appears.

**Testing and Reliability**

* Added comprehensive tests for the new dialogue condition logic, including quest state and skill level checks, and for Oak's dialogue flow to ensure correct visibility of choices as quests progress. [[1]](diffhunk://#diff-ed87e6e1e096a421aba8253028e6a469c0a8144ad5f9d2ba4f0dd88ddd4ee309R1-R66) [[2]](diffhunk://#diff-1e878289ceb6dd52762e3a1e4a65ed7666fbfb7068a31d5262829e48dde4bd38R1-R90)

**User Interface Improvements**

* Updated the dialogue UI to only display choices that are currently available based on the player's state, and to handle quest-related actions seamlessly within the dialogue. [[1]](diffhunk://#diff-12f59ceb7110735449b2c1ff896dedc406dc4f4832d95740300ecdab6a29b238L137-R184) [[2]](diffhunk://#diff-12f59ceb7110735449b2c1ff896dedc406dc4f4832d95740300ecdab6a29b238L61-R103)

These changes collectively make the dialogue and quest systems more dynamic, interactive, and aligned with player progression.

---
## Summary by cubic
Adds conditional NPC dialogue with quest triggers and moves tutorial quests to be started and completed through Oak. The quest list hides tutorials, and dialogue only shows choices that match the player’s state.

- **New Features**
  - Conditional dialogue choices based on quest state, objectives, skill levels, and inventory items.
  - Choices can start or complete quests directly; Oak’s dialogue drives the tutorial quest chain.
  - Tutorial quests are hidden from Available Quests and must be turned in via NPC dialogue.
  - Dialogue UI filters choices by conditions; QuestDetail disables direct completion for tutorial quests with guidance.
  - Added tests for condition logic and Oak’s dialogue flow; moved Oak to home for easier access.

<sup>Written for commit e75b214c6c9da2241afa8e829eaced8c5ebcca76. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Dialogue choices now show/hide based on game state and can trigger quest start/complete actions; new Oak dialogue options and nodes added.
* **Refactor**
  * Dialogue UI now filters choices by condition, executes quest outcomes before advancing, and surfaces transient error messages.
* **UX**
  * Quest detail button behavior/text changes for tutorial quests.
* **Tests**
  * Added comprehensive tests for dialogue conditions and quest-driven dialogue flows.
* **Chores**
  * Updated NPC location mappings and added Home NPC association.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<h2>Greptile Overview</h2>

<h3>Greptile Summary</h3>

Adds a well-designed conditional dialogue system that dynamically shows/hides choices based on quest progress, and allows quest actions to be triggered directly from NPC conversations. Tutorial quests are now hidden from the general quest list and must be started/completed via Oak's dialogue.

**Key Changes:**
- Implemented `checkCondition` function supporting quest state, skill level, inventory, and objective completion checks
- Modified `DialogueScreen` to filter choices by conditions and execute quest actions before advancing dialogue
- Tutorial quests filtered from available quest list in `getAvailableQuests`
- Oak moved to Home location and dialogue tree updated with conditional quest start/complete options
- `QuestDetail` shows disabled button with guidance for tutorial quest completion
- Comprehensive test coverage for condition logic and Oak's dialogue flow

**Concerns:**
- All implementations are clean and well-tested with proper error handling

<h3>Confidence Score: 5/5</h3>

- This PR is safe to merge with minimal risk
- The implementation is thorough with comprehensive test coverage, proper error handling, and clean separation of concerns. All edge cases are handled correctly in the condition checking logic, and the quest action flow properly validates before updating state.
- No files require special attention

<h3>Important Files Changed</h3>

File Analysis

| Filename | Score | Overview |
|----------|-------|----------|
| src/game/core/dialogue.ts | 5/5 | Added `checkCondition` function and quest action handling to `selectChoice`, enabling conditional dialogue choices based on game state |
| src/components/npc/DialogueScreen.tsx | 5/5 | Updated to filter choices by conditions, execute quest actions before advancing dialogue, and display error messages |
| src/components/quests/QuestDetail.tsx | 5/5 | Modified complete button for tutorial quests to show disabled state with guidance to return to Oak |
| src/game/core/quests/quests.ts | 5/5 | Added filter to hide tutorial quests from available quest list |
| src/game/data/dialogues.ts | 5/5 | Expanded Oak's dialogue tree with conditional choices for starting and completing tutorial quests |
| src/game/types/npc.ts | 5/5 | Added types for dialogue conditions and actions with comprehensive comparison operators |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Player
    participant DialogueScreen
    participant dialogue.ts
    participant GameState
    participant quests.ts
    
    Player->>DialogueScreen: Select dialogue choice
    DialogueScreen->>dialogue.ts: selectChoice(state, index, gameState)
    dialogue.ts->>dialogue.ts: checkCondition(gameState, conditions)
    
    alt Conditions not met
        dialogue.ts-->>DialogueScreen: {success: false, message: "Conditions not met"}
        DialogueScreen-->>Player: Show error message
    else Conditions met
        dialogue.ts->>dialogue.ts: Get next node
        dialogue.ts-->>DialogueScreen: {success: true, action, nextNode}
        
        alt Has quest action
            DialogueScreen->>quests.ts: startQuest() or completeQuest()
            
            alt Quest action fails
                quests.ts-->>DialogueScreen: {success: false, message}
                DialogueScreen-->>Player: Show error message (3s timeout)
            else Quest action succeeds
                quests.ts->>GameState: Update quest state
                quests.ts-->>DialogueScreen: {success: true, newState}
                DialogueScreen->>GameState: updateState()
                DialogueScreen->>DialogueScreen: Advance to next node
                DialogueScreen-->>Player: Show next dialogue
            end
        else No action
            DialogueScreen->>DialogueScreen: Advance to next node
            DialogueScreen-->>Player: Show next dialogue
        end
    end
```

**Context used:**

- Context from `dashboard` - AGENTS.md ([source](https://app.greptile.com/review/custom-context?memory=572ffb98-977a-4f2c-833b-573c17e24ccc))

### CODE REVIEW

File: src/components/quests/QuestDetail.tsx [hunk 1]
```diff
@@ -129,11 +130,17 @@ export function QuestDetail({
...
-          {canComplete && onComplete && (
-            <Button onClick={onComplete} className="w-full">
-              Complete Quest
-            </Button>
-          )}
+          {canComplete &&
+            onComplete &&
+            (quest.type === QuestType.Tutorial ? (
+              <Button disabled className="w-full" variant="outline">
+                Return to Oak to complete
```
[the remark this answers is not in the document]
Comment (KnightNiwrem) [reply src/components/quests/QuestDetail.tsx:137]: Accepted

File: AGENTS.md [hunk 2]
```diff
@@ -159,6 +159,6 @@ Follow this workflow when implementing new features:
 6. **Evaluate each review comment** and determine if it's valid
 7. **Address valid issues** with code changes
 8. **Commit and push** to the current branch
-9. **Request re-review** from Copilot using the `gh pr edit` with the `--add-reviewer` flag
+9. **Request re-review** from Copilot using the `github` MCP server.
```
[the remark this answers is not in the document]
Comment (KnightNiwrem) [reply AGENTS.md:162]: Rejected.
